File: src/org/stormgears/utils/dsio/XboxJoystick.java
package org.stormgears.utils.dsio;

import edu.wpi.first.wpilibj.Joystick;

public class XboxJoystick extends Joystick implements IRawJoystick {
	/**
	 * Construct an instance of a joystick. The joystick index is the USB port on the drivers
	 * station.
	 *
	 * @param port The port on the Driver Station that the joystick is plugged into.
	 */
	public XboxJoystick(int port) {
		super(port);
	}

	@Override
	public final double getJoystickX() {
		return getRawAxis(0);
	}

	@Override
	public double getJoystickY() {
		return getRawAxis(5);
	}

	@Override
	public double getJoystickZ() {
		return getRawAxis(4);
	}
}

File: src/main/java/org/stormgears/utils/dsio/XboxJoystick.kt
package org.stormgears.utils.dsio

import edu.wpi.first.wpilibj.Joystick

class XboxJoystick
/**
 * Construct an instance of a joystick. The joystick index is the USB port on the drivers
 * station.
 *
 * @param port The port on the Driver Station that the joystick is plugged into.
 */
(port: Int) : Joystick(port), IRawJoystick {

	override val joystickX: Double
		get() = getRawAxis(0)

	override val joystickY: Double
		get() = getRawAxis(5)

	override val joystickZ: Double
		get() = getRawAxis(4)
}
